src/function_lua.c

```diff
@@ -258,11 +306,44 @@ int luaEngineInitEngine() {
     lua_pcall(lua_engine_ctx->lua,0,1,0);
     lua_settable(lua_engine_ctx->lua, LUA_REGISTRYINDEX);
 
-    /* save the engine_ctx on the registry so we can get it from the Lua interpreter */
-    luaSaveOnRegistry(lua_engine_ctx->lua, REGISTRY_ENGINE_CTX_NAME, lua_engine_ctx);
+    /* Save global protection to registry */
```

Comment:
let's extract this to a function, and improve the doc comments

Reply:
Done